### DISCUSSION TITLE
Help with making code work in a different vault

### DISCUSSION THREAD

**Discussion by celiaccb** [Ji11ard/Dataview-to-Mermaid-Flowchart#8] [Q&A] [answered]
Hi,

Thanks for sharing this code! I have been trying to find a way to make dataview work in mermaid diagrams for a while. 

I am trying to use this in my own obsidian vault, but  I get the following error:

```
Evaluation Error: Error: Error: 
-- PARSING FAILED --------------------------------------------------

> 1 | TABLE WITHOUT ID author, "", "{{", "}}", "red" FROM Sources 📚/Books WHERE author FLATTEN author SORT author, genre, author
    |                                                     ^

Expected one of the following: 

'!', '(', '-', 'csv(', 'outgoing(', file link, string, tag ('#hello/stuff')

    at Failure.orElseThrow (plugin:dataview:8665:19)
    at DataviewApi.tryQuery (plugin:dataview:20121:65)
```

I was wondering if you could identify where have I gone wrong in the charting page YAML (see further below) ? 
Otherwise, could you just show an example of how the YAML in the files you are querying looks like? (I checked your example vault, but I am not sure it works with this specific repository?), I think that might
help me to understand how I need to change my YAML.

This is how the YAML of each file in the Sources 📚/Books folder I am trying to query looks like,
each file is a different book:

```
---
author: authorname
status: read
genre: realism, other
---
```

This is the charting page YAML:

```
---
Direction: "LR"
ShowCode: false
SubGroupNames: ["author", "read", "to-read"]
RemoveOrphans: false
KeepLinksWithoutSource: true
KeepLinksWithoutDest: true

Nodes:

 - 'TABLE WITHOUT ID author, "", "{{", "}}", "red" FROM Sources 📚/Books WHERE author FLATTEN author SORT author, genre, author'
 - 'TABLE file.name, "([", "])", "yellow" FROM Sources 📚/Books WHERE status = "read" SORT genre, author'
 - 'TABLE file.name, "([", "])", "aqua" FROM Sources 📚/Books WHERE status = "to-read" SORT genre, author'


Links:  
 - 'TABLE WITHOUT ID author, file.name, "---->" FROM Sources 📚/Books WHERE author FLATTEN author'

Styles: 
 - 'classDef Custom1 fill:#DDEEFF,color:#000,stroke:#000,stroke-width:1px'
---
```

My desired outcome would be a container on the left 'author' in which the nodes are the 'author' YAML field of each of the book files, with an arrow connecting them to the corresponding books on the right, which would be divided in two containers, one 'read' and the other 'to-read' (just like 'fiction' and 'non-fiction' in your example).

Thank you for any help!

CC

**Comment by Ji11ard** [comment, marked as answer]
Hey there!  I think it may be as simple as putting your FROM folder inside double quotes. 

Like in normal DQL queries, folder names go in quotes like a string, but tags stay naked. I think that's probably especially true when the folder name has spaces and slash characters in it.

Other than that, your YAML looks good.  (I've been meaning to add a version to the example vault, but just haven't gotten there yet).

**Reply by celiaccb** [reply]
Thank you! You were absolutely right and the graph looks great.

**Comment by celiaccb** [comment]
Hi, just one more question, the graph returned looks like this: 

<img width="715" alt="Screenshot 2022-10-12 at 09 40 50" src="https://user-images.githubusercontent.com/59609780/195294739-83b604e9-d807-4969-bf7f-1e9a506b6dda.png">

I was expecting there to be a third subgraph for 'genre', why is this not returned?

Thanks

**Reply by Ji11ard** [reply]
So at the moment you have three sets of nodes (author, read, to-read).  You'd need another query to get all the genre nodes (if you want them colored specially), and/or another link query to define the lines from books to to genres. 

Let me know if that makes sense, otherwise I can provide more details.... The example image in the readme is from my personal vault; where I have 4 node queries (author, fiction, nonf, topics), and 2 link queries (author-books, book-topics).

**Reply by celiaccb** [reply]
That makes sense!I just fixed it, thanks
